sync/p2p_messages.md

```diff
@@ -80,14 +80,33 @@ Message is RLP encoded, fields:
 
 Message is RLP encoded, fields:
 
-  - `Port :: int` - listen port
-  - `Share :: int` - number of peers to share
+### Version 1
+
+  - `Port :: int` - listen port.
+  - `Share :: int` - number of peers to share.
+  - `GenesisHash :: byte_array`
+  - `Difficulty :: int` - the total difficulty of the chain.
+  - `TopHash :: byte_array`
+  - `SyncAllowed :: bool` - if the sender of this ping message is accepting
+    synchronization messages.
+  - `Peers :: [byte_array]` - list of shared peers.
+
+
+### Version 2
+
+  - `Versions :: list`
+    - `Protocol :: binary` - type of message.
+    - `Vsns :: [int]` - versions supported
+  - `Port :: int` - listen port.
+  - `Share :: int` - number of peers to share.
   - `GenesisHash :: byte_array`
+  - `Height :: int` - current height.
```

Comment:
Isn't height kind of "per database" (node) field? Why is it shared? If you know the remote TopHash isn't it possible to derive its height as well?

Reply:
I asked Ulf and this is his response, "It's true that the height can be derived, but only by first downloading headers. While working on fast synch strategies, I felt that getting the height up front was a convenience at fairly low cost."